SubArraySumEqualsK.java

```diff
@@ -0,0 +1,46 @@
+/* Leetcode Problem - 560
+https://leetcode.com/problems/subarray-sum-equals-k/
+ Time Complexity : O(n)
+ Space Complexity : O(1)
```

Comment:
O(n), The length of the input array is not fixed hence the running sum and space of hashmap is not constant.